system/security/plugins/jwtSecurity/jwtSecurity.cpp

```diff
@@ -1013,7 +1038,8 @@ class CJwtSecurityManager : implements IDaliLdapConnection, public CBaseSecurity
 
 
 private:
-    mutable CriticalSection     crit;                       //!< Protects access to secretsName contents
+    mutable CriticalSection     jwtCrit;                    //!< Protects access to JWT secret file contents
+    mutable CriticalSection     daliFetchCrit;              //!< Protects access to Dali's key/value store
```

Comment:
Is this being used properly? It's unclear why fetching from the store on 972 requires protection, when deleting from the store on 172 does not. Also unclear is whether it is needed to protect access to the user cache when other cache updates are not similarly guarded.

Reply:
daliFetchCrit is used to serialize access to IDaliLdapConnection::getPermissions. It is not used as a value lock.

Two different gUserCache objects are instantiated within Dali, apparently one for feature requests and one for scoping checks (discovered while trying to make clearPermissions work).  Both gUserCache objects, however, will fetch, unpack, and verify identical data from Dali if possible, simultaneously.  The CriticalSection is there to ensure that the later one uses the just-cached user data from the fetch/unpack/validate.

I will add some explanatory comments to getPermissions().